src/pg/sql/11_kmeans.sql

```diff
@@ -1,21 +1,40 @@
-CREATE OR REPLACE FUNCTION  CDB_KMeans(query text, no_clusters integer,no_init integer default 20)
+-- Spatial k-means clustering
+
+CREATE OR REPLACE FUNCTION CDB_KMeans(query text, no_clusters integer, no_init integer default 20)
 RETURNS table (cartodb_id integer, cluster_no integer) as $$
-    
+
     from crankshaft.clustering import kmeans
-    return kmeans(query,no_clusters,no_init)
+    return kmeans(query, no_clusters, no_init)
+
+$$ LANGUAGE plpythonu;
 
-$$ language plpythonu;
+-- Non-spatial k-means clustering
+-- query: sql query to retrieve all the needed data
+
+CREATE OR REPLACE FUNCTION CDB_KMeansNonspatial(
+  query TEXT,
+  colnames TEXT[],
+  num_clusters INTEGER,
+  id_col TEXT DEFAULT 'cartodb_id',
+  standarize BOOLEAN DEFAULT true
+)
+RETURNS TABLE(cluster_label text, cluster_center text, rowid bigint) AS $$
```

Comment:
I was thinking of stringifying it anyway since the data type mappings between sql and plpython are pretty limited, so will be doing what you suggest @talos